### PR TITLE
HL-2 [FEATURE] Implementing RX CAN device

### PR DESCRIPTION
**Pull request by gilsllendo**
The can_filter_init method added in order to receive all the message from the CAN Bus, because in our case there is no more additional devices and any potential device should receive ach message from TX node.

### CODE REVIEW

**Comment by gilsllendo** [comment]
Hi @0x2A9 , according to this PR, I guess we need to move GPIO Pins related to LEDs to another definition, to know the correlation between pin number and a color.

File: src/drivers/can.c [hunk 1]
```diff
@@ -23,6 +23,41 @@ uint8_t can_init()
...
+    filter.CAN_FilterIdHigh = 0;
+    filter.CAN_FilterIdLow = 0;
+    filter.CAN_FilterMaskIdHigh = 0;
+    filter.CAN_FilterMaskIdLow = 0;
+    filter.CAN_FilterScale = CAN_FilterScale_16bit;
+
+    CAN_FilterInit(&filter);
+}
+
+uint8_t can_read(CAN_TypeDef* canx, uint8_t fifo_number, CanRxMsg* msg)
```
Comment (0x2A9) [inline src/drivers/can.c:43]: Let's use only one approach to specify the name of the port parameter. `canx` includes only lower case characters here, hovewer [there](https://github.com/lamia-mortis/led-headlights/blob/main/src/drivers/i2c.c#L21) already is another parameter with both upper and lower case characters.

Also please specify, which style is better in your opinion?

File: src/rx/main.c [hunk 2]
```diff
@@ -2,18 +2,31 @@
...
 
 int main(void)
 {
     gpio_output_mode_init();
     gpio_i2c_mode_init();
     i2c1_init();
+    gpio_can_mode_init();
     pca9685_init(PCA9685_DEFAULT_FREQUENCY);
+    
+    if (CAN_InitStatus_Success == can_init()) {
```
Comment (0x2A9) [inline src/rx/main.c]: Maybe we will use the same sequence of operands as [here](https://github.com/lamia-mortis/led-headlights/blob/main/src/tx/main.c#L14) during comparing operation?

**Comment by 0x2A9** [comment]
> Hi @0x2A9, according to this PR, I guess we need to move GPIO Pins related to LEDs to another definition, to know the correlation between pin number and a color.

I like this idea. Still it is better to implement that in a separate PR, though. We have tickets for refactoring as well.

File: src/drivers/can.c (hunk 1 above)
Comment (gilsllendo) [reply src/drivers/can.c:43]: Please, check the [ticket](https://txrx.atlassian.net/browse/HL-5?focusedCommentId=10002).

**Review by 0x2A9 — APPROVED** [review]

**Pull request merged**